iac-extensions/kubernetes/src/test/java/org/sonar/iac/helm/tree/GoAstTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.helm.tree;
+
+import com.google.protobuf.Any;
+import java.io.File;
+import java.io.IOException;
+import java.util.stream.Stream;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.sonar.iac.helm.tree.api.ActionNode;
+import org.sonar.iac.helm.tree.api.CommandNode;
+import org.sonar.iac.helm.tree.api.FieldNode;
+import org.sonar.iac.helm.tree.api.ListNode;
+import org.sonar.iac.helm.tree.utils.GoTemplateAstConverter;
+import org.sonar.iac.helm.utils.GoAstSupplier;
+
+import static org.sonar.iac.common.testing.IacTestUtils.code;
+
+class GoAstTest {
```

Comment:
I'm not sure what class tests this test, `GoTemplateAstConverter` or `Tree` or `HelmEvaluator`? Maybe it should be split?
This name will be hard to find in IntelliJ using shortcuts (jump to/from test).

Reply:
Yes, it could be split, and now it's testing several components at once. But I thought that it's better to operate on real data here, which can be produced by HelmEvaluator also here.